includes/Core/Modules/Module_Sharing_Settings.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Modules\Module_Sharing_Settings
+ *
+ * @package   Google\Site_Kit\Core\Modules
+ * @copyright 2022 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Modules;
+
+use Google\Site_Kit\Core\Storage\Setting;
+
+/**
+ * Base class for module sharing settings.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Module_Sharing_Settings extends Setting {
+
+	const OPTION = 'googlesitekit_dashboard_sharing';
+
+	/**
+	 * Gets the default value.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array
+	 */
+	protected function get_default() {
+		return array();
+	}
+
+	/**
+	 * Gets the expected value type.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return string The type name.
+	 */
+	protected function get_type() {
+		return 'object';
+	}
+
+	/**
+	 * Gets the callback for sanitizing the setting's value before saving.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return callable|null
+	 */
+	protected function get_sanitize_callback() {
+		return function( $option ) {
+			if ( ! is_array( $option ) ) {
+				return array();
+			}
+			$sanitized_option = array();
+			foreach ( $option as $module_slug => $sharing_settings ) {
+				$sanitized_option[ $module_slug ]['sharedRoles'] = $this->sanitize_string_list( $sharing_settings['sharedRoles'] );
+
+				if ( isset( $sharing_settings['management'] ) ) {
+					$sanitized_option[ $module_slug ]['management'] = (string) $sharing_settings['management'];
+				}
+			}
+
+			return $sanitized_option;
+		};
+	}
+
+	/**
+	 * Filter empty or non-string elements from a given array.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $elements Array to check.
+	 * @return array Empty array or a filtered array containing only non-empty strings.
+	 */
+	private function sanitize_string_list( $elements = array() ) {
+		if ( ! is_array( $elements ) ) {
+			$elements = array( $elements );
+		}
+
+		if ( empty( $elements ) ) {
+			return array();
+		}
+
+		$filtered_elements = array_filter(
+			$elements,
+			function( $element ) {
+				return is_string( $element ) && ! empty( $element );
+			}
+		);
+		// Avoid index gaps for filtered values.
+		return array_values( $filtered_elements );
+	}
+
+	/**
+	 * Gets the settings after filling in default values.
+	 *
+	 * {@inheritDoc}
+	 *
+	 * @since n.e.x.t
+	 */
+	public function get() {
+		$settings = parent::get();
+
+		foreach ( $settings as $module_slug => $sharing_settings ) {
+			if ( ! isset( $sharing_settings['management'] ) || ! in_array( $sharing_settings['management'], array( 'all_admins', 'owner' ), true ) ) {
+				$settings[ $module_slug ]['management'] = 'owner';
+			}
+		}
```

Comment:
@jimmymadon this is great but please also fill in the `sharedRoles` as well.

Reply:
I thought that since we were sanitizing `sharedRoles` when saving them and adding a default empty `array()`, we wouldn't need to add defaults when fetching. But makes sense to be consistent and safe here. 👍